### PR TITLE
[Fix] Fix recent build problems

### PR DESCRIPTION
**Pull request by lzhangzz**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by lvhan028** [comment]
I found the following block doesn't work on suppressing `CMAKE_CUDA_ARCHITECTURES` warning
```
if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18.0")
    # suppress 'CMAKE_CUDA_ARCHITECTURES' warning
    cmake_policy(SET CMP0104 OLD)
endif ()
```

The warning log is:
```
Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
  empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
  for policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.
```

Any clues?

**Comment by lvhan028** [comment]
@AllentDan Please check if this PR can fix the build issue you ran into in docker image

**Comment by lzhangzz** [comment]
> I found the following block doesn't work on suppressing `CMAKE_CUDA_ARCHITECTURES` warning
> 
> ```
> if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18.0")
>     # suppress 'CMAKE_CUDA_ARCHITECTURES' warning
>     cmake_policy(SET CMP0104 OLD)
> endif ()
> ```
> 
> The warning log is:
> 
> ```
> Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
>   empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
>   for policy details.  Use the cmake_policy command to set the policy and
>   suppress this warning.
> ```
> 
> Any clues?

This is because we have `cmake_minimum_required` command in all of our subdirectories, which is redundant and responsible for those warnings. 

We set `CMP0104` to `OLD` at the top-level. However, a `cmake_policy` command is implied by `cmake_minimum_required`, which will override the policy stack inherited from parent scope. At this point `CMP0104` is reset to an unset state and need to be set again to avoid the warning.

See https://gitlab.kitware.com/cmake/community/-/wikis/doc/cmake/Policies#policy-stack

**Comment by AllentDan** [comment]
> @AllentDan Please check if this PR can fix the build issue you ran into in docker image

It works fine now.

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,21 +19,28 @@ endif ()
...
-option(MMDEPLOY_BUILD_MODEL_OPTIMIZER "build MMDeploy's model optimizer" ON)
-option(MMDEPLOY_BUILD_SDK "build MMDeploy's SDK" OFF)
-option(MMDEPLOY_ZIP_MODEL "support sdk model in zip format" OFF)
-option(MMDEPLOY_BUILD_SDK_CSHARP_API "build MMDeployExtern.dll for C# API" OFF)
-option(MMDEPLOY_BUILD_TEST "build MMDeploy's csrc's unittest" OFF)
-set(MMDEPLOY_TARGET_DEVICES
-        "cpu" CACHE STRING "MMDeploy's target devices")
-set(MMDEPLOY_TARGET_BACKENDS "" CACHE STRING "MMDeploy's target inference engines")
-set(MMDEPLOY_CODEBASES "all"  CACHE STRING "select OpenMMLab's codebases")
+option(MMDEPLOY_SHARED_LIBS "build shared libs" ON)
```
Comment (lvhan028) [inline CMakeLists.txt:22]: May change `BUILD_SHARED_LIBS` to `MMDEPLOY_SHARED_LIBS` in `MMDeployConfig.cmake.in` accordingly.

**Comment by lvhan028** [comment]
Build documents should be updated, including removing external spdlog installation and changing `BUILD_SHARED_LIBS` to `MMDEPLOY_SHARED_LIBS`

**Review by lvhan028 — APPROVED** [review]

**Review by irexyc — APPROVED** [review]
test ok on windows

**Pull request merged**